File: PakingLotDemo/src/factories/PaymentFactory.java
package factories;

import enums.PaymentMode;
import strategy.payment.CardPaymentStrategy;
import strategy.payment.CashPaymentStrategy;
import strategy.payment.PaymentStrategy;

/**
 * Factory for obtaining the payment strategy for a given payment mode.
 */
public class PaymentFactory {

    /**
     * Returns the payment strategy for the given mode.
     *
     * @param mode the payment mode (e.g. CASH, CARD)
     * @return the strategy implementation
     * @throws IllegalArgumentException if mode is null or unknown
     */
    public static PaymentStrategy get(PaymentMode mode) {
        if (mode == null) {
            throw new IllegalArgumentException("PaymentMode cannot be null");
        }
        return switch (mode) {
            case CASH -> new CashPaymentStrategy();
            case CARD -> new CardPaymentStrategy();
            default -> throw new IllegalArgumentException("Unknown PaymentMode: " + mode);
        };
    }
}

File: problems/python/ParkingLotSystem/src/factories/payment_factory.py
"""Factory for payment strategy by payment mode."""
from enums.payment_mode import PaymentMode
from strategy.payment.payment_strategy import PaymentStrategy
from strategy.payment.cash_payment_strategy import CashPaymentStrategy
from strategy.payment.card_payment_strategy import CardPaymentStrategy


class PaymentFactory:
    @staticmethod
    def get(mode: PaymentMode) -> PaymentStrategy:
        if mode is None:
            raise ValueError("PaymentMode cannot be null")
        if mode == PaymentMode.CASH:
            return CashPaymentStrategy()
        if mode == PaymentMode.CARD:
            return CardPaymentStrategy()
        raise ValueError(f"Unknown PaymentMode: {mode}")
